cou/steps/plan.py

```diff
@@ -596,7 +595,27 @@ async def _get_upgradable_hypervisors_machines(
     if cli_force:
         return nova_compute_machines
 
-    return await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+    empty_hypervisors = await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+
+    machine_to_unit_name = {
+        machine.machine_id: unit.name
+        for unit, machine in zip(nova_compute_units, nova_compute_machines)
+    }
+    empty_hypervisor_ids = {machine.machine_id for machine in empty_hypervisors}
+    all_hypervisor_ids = {machine.machine_id for machine in nova_compute_machines}
+    skipped_hypervisor_ids = all_hypervisor_ids - empty_hypervisor_ids
+
+    skipped_unit_names = ", ".join(
+        machine_to_unit_name[machine_id] for machine_id in sorted(skipped_hypervisor_ids)
+    )
+    logger.info("Skipped (non-empty) hypervisors: %s", skipped_unit_names)
+
+    selected_unit_names = ", ".join(
+        machine_to_unit_name[machine_id] for machine_id in sorted(empty_hypervisor_ids)
+    )
+    logger.info("Selected (empty) hypervisors: %s", selected_unit_names)
```

Comment:
see how lines 614 and 609 are doing the same thing and can be considered code of lower level than the other lines. I suggest creating a helper function and then use something like:

```python
logger.info("Selected (empty) hypervisors: %s", _stringify_units(selected_unit_names))
```

Reply:
Should I put the helper function under utils/nova_compute.py ?

Reply:
you could use a lambda for such a small helper